ayaneo-platform.c

```diff
@@ -422,73 +368,140 @@ static void ayaneo_led_mc_set(u8 group, u8 pos, u8 brightness)
                 return;
 }
 
-static void ayaneo_led_mc_release(void) {
+static void ayaneo_led_mc_legacy_release(void)
+{
         if (!lock_global_acpi_lock())
                 return;
 
         ec_write(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_RELEASE);
 
         if (!unlock_global_acpi_lock())
                 return;
+}
+
+static void ayaneo_led_mc_legacy_hold(void)
+{
+        if (!lock_global_acpi_lock())
+                return;
 
+        ec_write(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_HOLD);
+
+        if (!unlock_global_acpi_lock())
+                return;
 }
 
-static void ayaneo_led_mc_intensity(u8 *color, u8 group, u8 zones[])
+static void ayaneo_led_mc_legacy_intensity(u8 group, u8 *color, u8 zones[])
 {
         int zone;
 
         for (zone = 0; zone < 4; zone++) {
-                ayaneo_led_mc_set(group, zones[zone], color[0]);
-                ayaneo_led_mc_set(group, zones[zone] + 1, color[1]);
-                ayaneo_led_mc_set(group, zones[zone] + 2, color[2]);
+                ayaneo_led_mc_legacy_set(group, zones[zone], color[0]);
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 1, color[1]);
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 2, color[2]);
         }
+        ayaneo_led_mc_legacy_set(0x03, 0x00, 0x00);
```

Comment:
What does this additional set call do?

Reply:
I believe it signals the microcontroller to apply the change, kind of like a commit or CTS.

0x03 refers to both groups (0x01 & 0x02), left and right.

Without it I noticed the on/off commands occasionally failed to apply prior to suspend.